pkg/controller/jobframework/reconciler.go

```diff
@@ -365,7 +366,29 @@ func (r *JobReconciler) equivalentToWorkload(job GenericJob, object client.Objec
 	if owner.Name != object.GetName() {
```

Comment:
There are two problems with this:
- this is called only on workload object from within the same namespace
- there is no namespace in the ownerRef.
We could change it to check the UID, but it will be harder to follow and the added value will be small in my opinion.

Reply:
Valid points.